cypress/integration/common/kiali_login.ts

```diff
@@ -0,0 +1,52 @@
+import { Given, When, And, Then } from "cypress-cucumber-preprocessor/steps";
+
+const USERNAME = 'jenkins';
+const PASSWD = Cypress.env('passwd')
+const KUBEADMIN_IDP = 'my_htpasswd_provider';
+const auth_strategy = Cypress.env('auth_strategy')
+
+Given('user opens base url', () => {
+    cy.visit('/');
+    cy.log(auth_strategy)
+    cy.window().then((win: any) => {
+        if (auth_strategy != 'openshift') {
+            cy.log('Skipping login, Kiali is running with auth disabled');
+        }
+        // Make sure we clear the cookie in case a previous test failed to logout.
+        cy.clearCookie('openshift-session-token');
+    });
+});
+
+And('user clicks Log In With OpenShift', () => {
+    if (auth_strategy === 'openshift') {
+        const idp = KUBEADMIN_IDP;
+        cy.log(`Logging in as ${USERNAME}`);
+        cy.get('button[type="submit"]').should('be.visible');
+        cy.get('button[type="submit"]').click();
```

Comment:
We talked offline about this but cypress hangs here if the auth redirect domain does not match the same domain as the app. We don't need to decide how we test third party auth integrations in this PR but something to keep in mind for the future.